test/utilities/testsUtils.ts

```diff
@@ -0,0 +1,224 @@
+ * );
+ * console.log(result); // Outputs the result from the Apex test run
+ */
+export async function runTestCaseFromSideBar(
+  workbench: Workbench,
+  testSuite: string,
+  testName: string,
+  actionLabel: string
+): Promise<string | undefined> {
+  log(`Running ${testSuite} - ${testName} - ${actionLabel} from SideBar`);
+  const testingView = await workbench.getActivityBar().getViewControl('Testing');
+  await expect(testingView).not.toBeUndefined();
+
+  // Open the Test Sidebar
+  const testingSideBarView = await testingView?.openView();
+  await expect(testingSideBarView).toBeInstanceOf(SideBarView);
+
+  // Select test
+  const testSection = await getTestsSection(workbench, testSuite);
+  const testItem = (await testSection.findItem(testName)) as TreeItem;
+  await expect(testItem).toBeDefined();
+  await testItem.select();
+
+  // Click button to run test
+  const actionButton = await testItem.getActionButton(actionLabel);
+  await expect(actionButton).toBeDefined();
+  await actionButton?.elem.click();
+
+  if (actionLabel.includes('Debug')) {
+    await pause(Duration.seconds(10));
+
+    // Continue with the debug session
+    await continueDebugging();
+  }
+
+  let testResult: string | undefined;
+  if (testSuite === 'APEX TESTS') {
+    // Look for the success notification that appears which says, "SFDX: Run Apex Tests successfully ran".
+    const successNotificationWasFound = await notificationIsPresentWithTimeout(
+      'SFDX: Run Apex Tests successfully ran',
+      Duration.TEN_MINUTES
+    );
+    await expect(successNotificationWasFound).toBe(true);
+    testResult = await attemptToFindOutputPanelText('Apex', '=== Test Results', 10);
+  } else if (testSuite === 'LWC TESTS') {
+    testResult = await getTerminalViewText(workbench, Duration.seconds(15));
+  }
+  await verifyTestIconColor(testItem, 'testPass');
+  return testResult;
+}
+
+/**
+ * Verifies the color of the test icon in the sidebar to ensure it reflects the correct test status.
+ *
+ * @param {TreeItem} testItem - The test item whose icon color needs to be verified. It represents a node in the sidebar tree view.
+ * @param {string} colorLabel - The expected color label (e.g., 'testPass', 'testNotRun') that indicates the test status.
+ *
+ * @example
+ * await verifyTestIconColor(myTestItem, 'testPass'); // Verifies the icon is green for a passing test
+ */
+export async function verifyTestIconColor(testItem: TreeItem, colorLabel: string) {
+  log(`Verifying icon's colors - verifyTestIconColor()`);
+  // Verify the tests that are passing are labeled with a green dot on the Test sidebar
+  const icon = await (await testItem.elem).$('.custom-view-tree-node-item-icon');
+  const iconStyle = await icon.getAttribute('style');
+  // Try/catch used to get around arbitrary flaky failure on Ubuntu in remote
+  try {
+    await expect(iconStyle).toContain(colorLabel);
+  } catch {
+    log('ERROR: icon did not turn green after test successfully ran');
```

Comment:
```suggestion
    log(`ERROR: icon color is not ${colorLabel}`);
```

Also, I see the above comment that the try/catch was used to get around an arbitrary flaky failure in Ubuntu.  Can we verify whether the arbitrary flaky failure still exists and remove the try/catch if it's no longer an issue?

Reply:
I think we should keep it. We have not done any changes around that and as it says, it's flaky. Maybe it will pass in a couple of runs and then just start failing again, and we don't want to introduce more flappers :dead1: